pkg/controllers/nodeclaim/lifecycle/liveness.go

```diff
@@ -51,6 +56,12 @@ func (l *Liveness) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (reco
 	if ttl := registrationTTL - l.clock.Since(registered.LastTransitionTime.Time); ttl > 0 {
 		return reconcile.Result{RequeueAfter: ttl}, nil
 	}
+	if err := l.updateNodePoolRegistrationHealth(ctx, nodeClaim); err != nil {
+		if errors.IsConflict(err) {
```

Comment:
nit: It's a little odd to send the conflict error back but not send the NotFound error back -- honestly, I would just handle both out here but it's your call